### ISSUE TITLE
`excepted` calculations but Job State still WITHSCHEDULER

### ISSUE THREAD

**Issue by giovannipizzi** [aiidateam/aiida_core#1687]
I've got a calculation that ended up in this state:
`15204  26m ago     Excepted | None  quantumespresso.pw  daint-mc    WITHSCHEDULER | None`
shouldn't the state be set to some other state if the calculation is Excepted?

Or even more specifically, in this case, shouldn't there be a way to revert back from the excepted state, and keep checking for the scheduler state? (The calculation is actually running, in my case the command to get the list of calcs in the queue had problems).

Note: the cause was that I changed some option in the scheduler (to allow SLURM to query by user) that triggered an internal exception, the full traceback is:
```
06/27/2018 10:01:05 AM <138392> tornado.application: [ERROR] Future exception was never retrieved: Traceback (most recent call last):
  File "/home/aiida/.virtualenvs/aiida_wannier_scdm/lib/python2.7/site-packages/tornado/gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "/home/aiida/.virtualenvs/aiida_wannier_scdm/lib/python2.7/site-packages/kiwipy/rmq/tasks.py", line 141, in _on_task
    result = yield task_future
  File "/home/aiida/.virtualenvs/aiida_wannier_scdm/lib/python2.7/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/home/aiida/.virtualenvs/aiida_wannier_scdm/lib/python2.7/site-packages/kiwipy/futures.py", line 43, in result
    return super(Future, self).result(timeout=0.)
  File "/home/aiida/.virtualenvs/aiida_wannier_scdm/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "/home/aiida/code/aiida/env/wannier_scdm/aiida_core/aiida/work/job_processes.py", line 236, in execute
    job_done = yield self._task
  File "/home/aiida/.virtualenvs/aiida_wannier_scdm/lib/python2.7/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/home/aiida/.virtualenvs/aiida_wannier_scdm/lib/python2.7/site-packages/kiwipy/futures.py", line 43, in result
    return super(Future, self).result(timeout=0.)
  File "/home/aiida/.virtualenvs/aiida_wannier_scdm/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "/home/aiida/code/aiida/env/wannier_scdm/aiida_core/aiida/work/job_processes.py", line 59, in _execute
    self.set_result(self.execute(transport))
  File "/home/aiida/code/aiida/env/wannier_scdm/aiida_core/aiida/work/job_processes.py", line 94, in execute
    found_jobs = scheduler.getJobs(**kwargs)
  File "/home/aiida/code/aiida/env/wannier_scdm/aiida_core/aiida/scheduler/__init__.py", line 343, in getJobs
    retval, stdout, stderr = self.transport.exec_command_wait(self._get_joblist_command(jobs=jobs, user=user))
  File "/home/aiida/code/aiida/env/wannier_scdm/aiida_core/aiida/scheduler/plugins/slurm.py", line 198, in _get_joblist_command
    raise FeatureNotAvailable("Cannot query by user and job(s) in SLURM")
FeatureNotAvailable: Cannot query by user and job(s) in SLURM
```

**Comment by sphuber** [comment]
Technically the right thing happened, it was the JobProcess that excepted, however, as you already noted, the Calculation is still running on the scheduler and so its `CalculationState` is actually correct. The fact that we have two states for a JobCalculation, the calculation state and the process state, makes it a bit difficult to understand all this but also to code. For example, `verdi calculation list` still used the `JobCalculation.state` attribute as leading in determining whether something is active, where anywhere else the process state is leading. I think this example is therefore indicative of a larger problem that we should probably discuss and fix.

Regarding reviving the process, I am not sure how easy that would be to accomplish. Currently the checkpoint gets deleted when the process terminates, so it would be impossible to recreate its state. In theory we can remove this logic and keep the checkpoint, in which case I think it should be possible in theory to add functionality to attempt to recreate a process task and have the daemon pick it up where it left off. This is assuming that the checkpoint would not be corrupted of course

**Comment by sphuber** [comment]
Note that this is a special case of a more general bug that will be addressed in #1698

**Comment by sphuber** [comment]
Exception in transport tasks of JobProcesses are now dealt with gracefully by the exponential backoff retry and pause mechanism, which is merged into `develop`.
